### ISSUE TITLE
'draw' fires twice on .update()

### ISSUE THREAD

**Issue by BumbleB2na** [chartist-js/chartist#206]
I think I've found 2 bugs in v0.7.2 (line chart updates):

'draw' event fires once if I do this:
myChart.update(myData);

Bug 1: 'draw' event fires twice if I do this:
myChart.update(myData, myOptions, true);

Bug 2: "Uncaught TypeError: Cannot read property 'chart' of undefined" on line 2443 in chartist.js if I choose to not extend the options:
myChart.update(myData, myOptions, false);   // or:
myChart.update(myData, myOptions);

Bug 2 probably related to bug 1..  because it only throws the error once, presumably on the second 'draw' attempt.

**Comment by gionkunz** [comment]
Thanks for the report! You're right there is a bug and unfortunately this is not a very well tested part of the library. 

There are actually two problems. The first one is that on optionsProvider recreation there is an immediate optionsChanged event triggered that will trigger an other update (thats why you've seen two events)

Actually the problem boils down to this line https://github.com/gionkunz/chartist-js/blob/develop/src/scripts/base.js#L41

The second problem, which of I didn't even think when I've accepted this PR for additional options override in updates, is that if you specify not to extend to options, an empty object will be used as base for the options. Which is completely non-sense as there are many default configurations required so Chartist can do its work. I think I've fixed these issues and will write some unit tests for this too.
